bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,873 @@
+
+        @Override
+        public void fatalError(Error error) {
+            owner.fatalError(this, error);
+        }
+
+        @Override
+        public void onClose(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalError(Error.CLOSED);
+        }
+
+        @Override
+        public void onFailure(@Nullable Session session, @Nullable Throwable failure) {
+            fatalError(Error.FAILURE);
+        }
+
+        @Override
+        public void onGoAway(@Nullable Session session, @Nullable GoAwayFrame frame) {
+            fatalError(Error.GOAWAY);
+        }
+
+        @Override
+        public boolean onIdleTimeout(@Nullable Session session) {
+            return false;
+        }
+
+        @Override
+        public void onReset(@Nullable Session session, @Nullable ResetFrame frame) {
+            fatalError(Error.RESET);
+        }
+    }
+
+    private static final String APPLICATION_ID = "org-openhab-binding-hue-clip2";
+    private static final String APPLICATION_KEY = "hue-application-key";
+
+    private static final String FORMAT_URL_CONFIG = "http://%s/api/0/config";
+    private static final String FORMAT_URL_RESOURCE = "https://%s/clip/v2/resource/";
+    private static final String FORMAT_URL_REGISTER = "http://%s/api";
+    private static final String FORMAT_URL_EVENTS = "https://%s/eventstream/clip/v2";
+
+    private static final int CLIP2_MINIMUM_VERSION = 1948086000;
+    private static final int TIMEOUT_SECONDS = 10;
+    private static final int CHECK_ALIVE_SECONDS = 300;
+
+    private static final ResourceReference BRIDGE = new ResourceReference().setType(ResourceType.BRIDGE);
+
+    /**
+     * Static method to attempt to connect to a Hue Bridge, get its software version, and check if it is high enough to
+     * support the CLIP 2 API.
+     *
+     * @param hostName the bridge IP address.
+     * @return returns without any exception if the bridge is online and it does support CLIP 2.
+     * @throws ApiException if was not possible to connect or another error was encountered.
+     * @throws IllegalArgumentException if the hostName is bad.
+     * @throws IllegalStateException if it was possible to connect but CLIP 2 is not supported.
+     */
+    public static void testSupportsClip2(String hostName)
+            throws ApiException, IllegalStateException, IllegalArgumentException {
+        //
+        HttpRequest request;
+        try {
+            request = HttpRequest.newBuilder().uri(new URI(String.format(FORMAT_URL_CONFIG, hostName)))
+                    .header(HttpHeader.ACCEPT.toString(), MediaType.APPLICATION_JSON)
+                    .timeout(Duration.ofSeconds(TIMEOUT_SECONDS)).build();
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Bad host name");
+        }
+        HttpResponse<String> response;
+        try {
+            response = java.net.http.HttpClient.newHttpClient().send(request, BodyHandlers.ofString());
```

Comment:
Yes. Because, as mentioned above, this is a STATIC method that is called (once only) before an thing would be allowed to be created.

Reply:
Perhaps [HttpUtil](https://www.openhab.org/javadoc/latest/org/openhab/core/io/net/http/httputil)'s static `executeUrl` methods would be more light-weight? At least it will use the common client:

https://github.com/openhab/openhab-core/blob/a09ad3d37c0e1fa5136c3a2e51f180e81886d97e/bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/HttpUtil.java#L194

Reply:
Ok.